app/src/main/java/ch/epfl/sdp/blindly/UserMapActivity.kt

```diff
@@ -0,0 +1,127 @@
+package ch.epfl.sdp.blindly
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import ch.epfl.sdp.blindly.permissions.LocationPermission.Companion.LOCATION_PERMISSION_REQUEST_CODE
+import ch.epfl.sdp.blindly.permissions.LocationPermission.Companion.PermissionDeniedDialog.Companion.newInstance
+import ch.epfl.sdp.blindly.permissions.LocationPermission.Companion.isPermissionGranted
+import ch.epfl.sdp.blindly.permissions.LocationPermission.Companion.requestPermission
+
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+
+/*
+ * Act
+ */
+class UserMapActivity: AppCompatActivity(), OnMapReadyCallback, ActivityCompat.OnRequestPermissionsResultCallback {
+    /**
+     * Flag indicating whether a requested permission has been denied after returning in
+     * [.onRequestPermissionsResult].
+     */
+    private var permissionDenied = false
+    private lateinit var map: GoogleMap
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.user_map_activity)
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        val mapFragment = supportFragmentManager
+                .findFragmentById(R.id.map) as SupportMapFragment?
+        mapFragment!!.getMapAsync(this)
+    }
+
+    /**
+     * Manipulates the map once available.
+     * This callback is triggered when the map is ready to be used.
+     * This is where we can add markers or lines, add listeners or move the camera. In this case,
+     * we just add a marker near Sydney, Australia.
+     *
+     * If Google Play services is not installed on the device, the user will be prompted to install
+     * it inside the SupportMapFragment. This method will only be triggered once the user has
+     * installed Google Play services and returned to the app.
+     */
+    override fun onMapReady(googleMap: GoogleMap?) {
+        map = googleMap ?: return
+
+        enableMyLocation()
+
+        // Add markers from the intent
+        val points: Array<LatLng>? = intent.getParcelableArrayExtra(POINTS)?.map { p -> p as LatLng }?.toTypedArray()
+        if (points != null) {
+            for (point in points) {
+                map!!.addMarker(MarkerOptions().position(point).title("Marker in Sydney2"))
+            }
+        }
+
+        // TODO move camera to user settings position
```

Comment:
You marked it as a TODO, but do we actually need the camera for the map?

Reply:
Oh yes, its working now indeed. I think I forgot it, thanks for the remark!

Reply:
By camera it is meant, having the user's position to the center of the map. However I don't get the user position for now, it is to be done when things are sent to firebase live.